diff --git a/src/main/java/org/springframework/samples/petclinic/owner/OwnerController.java b/src/main/java/org/springframework/samples/petclinic/owner/OwnerController.java
index 0bf6c7e..dbed53a 100644
--- a/src/main/java/org/springframework/samples/petclinic/owner/OwnerController.java
+++ b/src/main/java/org/springframework/samples/petclinic/owner/OwnerController.java
@@ -21,0 +22 @@ import java.util.Optional;
+import at.gepardec.fachtracing.api.FachTracing;
@@ -94,0 +96 @@ class OwnerController {
+	@FachTracing("owner search")
diff --git a/src/main/java/org/springframework/samples/petclinic/owner/VisitController.java b/src/main/java/org/springframework/samples/petclinic/owner/VisitController.java
index c832189..0f0a797 100644
--- a/src/main/java/org/springframework/samples/petclinic/owner/VisitController.java
+++ b/src/main/java/org/springframework/samples/petclinic/owner/VisitController.java
@@ -19,0 +20 @@ import java.util.Optional;
+import at.gepardec.fachtracing.api.FachTracing;
@@ -97,0 +99 @@ class VisitController {
+	@FachTracing("visit booking")
diff --git a/src/main/java/org/springframework/samples/petclinic/owner/PetController.java b/src/main/java/org/springframework/samples/petclinic/owner/PetController.java
index 845c116..20c4128 100644
--- a/src/main/java/org/springframework/samples/petclinic/owner/PetController.java
+++ b/src/main/java/org/springframework/samples/petclinic/owner/PetController.java
@@ -22,0 +23 @@ import java.util.Optional;
+import at.gepardec.fachtracing.api.FachTracing;
@@ -107,0 +109 @@ class PetController {
+	@FachTracing("pet registration")
